### ISSUE TITLE
Eldev erroneously treats its own script as an Eldev file on a case-insensitive file system

### ISSUE THREAD

**Issue by DarwinAwardWinner** [doublep/eldev#9]
Currently, I cannot install and use eldev. When I attempt to do so, I get this:

```
$ rm ~/.eldev eldev # Clean out any existing eldev
$ curl -fsSL https://raw.github.com/doublep/eldev/master/bin/eldev > eldev && chmod a+x eldev
$ eldev
Bootstrapping Eldev for Emacs 26.3 from Melpa stable...

Importing package-keyring.gpg...
Importing package-keyring.gpg...done
Invalid read syntax: "#"
```

Manually installing by cloning the repo and running `install.sh` does work.

**Comment by doublep** [comment]
I cannot reproduce this. Tried on two machines. Tried with Emacs 28 (trunk) and Emacs 26.3.

**Comment by DarwinAwardWinner** [comment]
Ok, it seems this only happens when I run eldev from the directory where it is located. E.g. if the path to `eldev` is `~/bin/eldev`, then:

```
$ pwd
/Users/ryan
$ eldev
Usage: eldev [OPTION...] COMMAND [...]
Run ‘eldev help’ for more information
$ cd ~/bin/
$ eldev
Invalid read syntax: "#"

**Comment by DarwinAwardWinner** [comment]
Oh, of course: the shell script file name `eldev` is being treated as an `Eldev` file, so eldev is trying to read its own shell source as elisp.

**Comment by DarwinAwardWinner** [comment]
It's possible this is only a problem on Mac OS with its default case-insensitive filesystem.

**Comment by doublep** [comment]
Well, it is kinda a corner-case then. But I guess I can just add a special case to avoid reading `Eldev` if the file starts with `#!`. It is too late to change filenames (i.e. `eldev` executable vs. `Eldev` as its build file).

**Comment by DarwinAwardWinner** [comment]
Another approach might be to record the path of the script bring executed and then not count that path as a valid Eldev file.

**Comment by doublep** [comment]
Fix will be released in 0.3.
